### PR TITLE
Add Upload book modal in publisher profile

### PR DESCRIPTION
**Pull request by subinasr**
addresses https://github.com/kitab-bazar/kitab-bazar/issues/22
- list publisher books in publisher profile

### CODE REVIEW

File: app/views/BookList/index.tsx [hunk 1]
```diff
@@ -185,7 +188,9 @@ function Explore(props: Props) {
         language?: string;
     } | undefined;
 
-    // const canAddBook = user?.permissions.includes('CAN_CREATE_BOOK');
+    const canPublisherCreateBook = isDefined(user?.publisherId);
```
Comment (subinasr) [inline app/views/BookList/index.tsx]: Let's change the variable name to isPublisher
```suggestion
    const isPublisher = isDefined(user?.publisherId);
```

File: app/views/BookList/index.tsx [hunk 2]
```diff
@@ -185,7 +188,9 @@ function Explore(props: Props) {
...
+    const canModeratorCreateBook = user?.type === 'MODERATOR';
```
Comment (subinasr) [inline app/views/BookList/index.tsx]:
```suggestion
    const isModerator = user?.type === 'MODERATOR';
```

File: app/components/BookItem/index.tsx [hunk 3]
```diff
@@ -159,6 +163,14 @@ function BookItem(props: Props) {
     const { user } = useContext(UserContext);
 
     const canCreateOrder = user?.permissions.includes('CREATE_ORDER');
+    const canPublisherUpdateBook = variant === 'detail' && book.publisher.id === user?.publisherId;
```
Comment (subinasr) [inline app/components/BookItem/index.tsx]:
```suggestion
    const isPublisher = variant === 'detail' && book.publisher.id === user?.publisherId;
```

File: app/components/BookItem/index.tsx [hunk 4]
```diff
@@ -159,6 +163,14 @@ function BookItem(props: Props) {
...
+    const canModeratorUpdateBook = user?.type === 'MODERATOR';
```
Comment (subinasr) [inline app/components/BookItem/index.tsx]:
```suggestion
    const isModerator = user?.type === 'MODERATOR';
```

**Review by samshara — APPROVED** [review]

**Pull request merged**